lib/ansible/modules/windows/win_git.ps1

```diff
@@ -0,0 +1,202 @@
+#!powershell
+
+# (c) 2017, Anatoliy Ivashina <tivrobo@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$params = Parse-Args $args;
+
+$result = New-Object psobject @{
+    win_git = New-Object psobject @{
+        name              = $null
+        dest              = $null
+        replace_dest      = $false
+        accept_hostkey    = $false
+    }
+    msg = $null
+    changed = $false
+}
+
+$name = Get-AnsibleParam -obj $params -name "name" -failifempty $true
+$dest = Get-AnsibleParam -obj $params -name "dest"
+$replace_dest = ConvertTo-Bool (Get-AnsibleParam -obj $params -name "replace_dest" -default $false)
+$accept_hostkey = ConvertTo-Bool (Get-AnsibleParam -obj $params -name "accept_hostkey" -default $false)
+
+$_ansible_check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
```

Comment:
Simply call this $check_mode instead. Keep it short and simple.

Make it of -type "bool".